### DISCUSSION TITLE
How to get existing exchange binding from a queue?

### DISCUSSION THREAD

**Discussion by joe-agent** [rabbitmq/rabbitmq-dotnet-client#1236] [General]
We have an existing queue called `QueueA` with a binding of `ExchangeA`

Now we need to remove the binding of `ExchangeA` from `QueueA` by calling `QueueUnbind` method

How do we get the current binding exchange name from `QueueA`?
Is there any public method with example available?
Thanks

**Comment by michaelklishin** [comment]
Bindings do not have any kind of ID, so `queue.unbind` uses a combination of source name (exchange name), destination name (in this case, queue name), routing key and (rarely used) optional binding arguments.

**Reply by joe-agent** [reply]
@michaelklishin , yes but the code where I call `queue.unbind` does not know what the exchange name is.

Is there any API method I can use from `rabbitmq-dotnet-client` to get all the existing exchange names bound to the queue name?

**Reply by michaelklishin** [reply]
Then you cannot unbind that queue. Bindings do not have IDs, all operations on them are performed using source and destination names.

**Comment by lukebakken** [comment]
@joe-agent you can use the HTTP API to get queue binding information:

https://rawcdn.githack.com/rabbitmq/rabbitmq-server/v3.13.2/deps/rabbitmq_management/priv/www/api/index.html
